### PR TITLE
remove examples gateway.

### PR DESCRIPTION
**Pull request by lkk12014402**
## Description

remove examples gateway after this pr https://github.com/opea-project/GenAIExamples/pull/992 merged

### CODE REVIEW

File: comps/cores/mega/gateway.py [hunk 1]
```diff
@@ -4,24 +4,12 @@
 import base64
```
Comment (ftian1) [inline comps/cores/mega/gateway.py:4]: shall you totally remove this file as comps shouldn't have gateway concept.

Comment (lkk12014402) [reply comps/cores/mega/gateway.py:4]: shall we move `gateway.py`  to `service.py`, and rename the class `Gateway`  to ```Service` or `BaseService`?

**Comment by dmsuehir** [comment]
@lkk12014402  Will the python gateway tests (such as https://github.com/opea-project/GenAIComps/blob/main/tests/cores/mega/test_multimodalqna_gateway.py) be moved to GenAIExamples?

File: comps/cores/mega/http_service.py [hunk 2]
```diff
@@ -137,6 +142,35 @@ async def terminate_server(self):
...
+
+    def run(self):
+        """Running method to block the main thread.
+
+        This method runs the event loop until a Future is done. It is designed to be called in the main thread to keep it busy.
+        """
+        self.event_loop.run_until_complete(self.execute_server())
+
+    def start(self, in_single_process=False):
+        print(in_single_process)
```
Comment (ftian1) [inline comps/cores/mega/http_service.py:158]: shouldn't use print statement here.

Comment (lkk12014402) [reply comps/cores/mega/http_service.py:158]: sorry forget removing

**Comment by lkk12014402** [comment]
> @lkk12014402 Will the python gateway tests (such as https://github.com/opea-project/GenAIComps/blob/main/tests/cores/mega/test_multimodalqna_gateway.py) be moved to GenAIExamples?

yes

File: comps/cores/mega/utils.py [hunk 3]
```diff
@@ -258,3 +261,73 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type:
             self.context_to_manage.__exit__(exc_type, exc_val, exc_tb)
+
+
+def _handle_message(messages):
```
Comment (ftian1) [inline comps/cores/mega/utils.py:266]: what's it for? from naming convention, it has an underscore prefix and act as a internal func. but nobody uses it?

Comment (lkk12014402) [reply comps/cores/mega/utils.py:266]: many GenAIExamples need the function

**Review by Spycsh — APPROVED** [review]

**Comment by dmsuehir** [comment]
> > @lkk12014402 Will the python gateway tests (such as https://github.com/opea-project/GenAIComps/blob/main/tests/cores/mega/test_multimodalqna_gateway.py) be moved to GenAIExamples?
> 
> hi, for the `multimodalqna` test, there is a ut https://github.com/opea-project/GenAIExamples/tree/main/MultimodalQnA/tests in the GenAIExamples

These are .sh script tests in GenAIExamples, but the ones in `test_multimodalqna_gateway.py` are python unit tests and it is stubbing out the other microservices in order to isolate testing the gateway, which I find pretty useful. I didn't see other python tests in GenAIExamples, so I wasn't sure if moving the .py test file to `https://github.com/opea-project/GenAIExamples/tree/main/MultimodalQnA/tests` would automatically work or if there would be more work associated with getting GHA to run those python test.

**Comment by lkk12014402** [comment]
> > > @lkk12014402 Will the python gateway tests (such as https://github.com/opea-project/GenAIComps/blob/main/tests/cores/mega/test_multimodalqna_gateway.py) be moved to GenAIExamples?
> > 
> > 
> > hi, for the `multimodalqna` test, there is a ut https://github.com/opea-project/GenAIExamples/tree/main/MultimodalQnA/tests in the GenAIExamples
> 
> These are .sh script tests in GenAIExamples, but the ones in `test_multimodalqna_gateway.py` are python unit tests and it is stubbing out the other microservices in order to isolate testing the gateway, which I find pretty useful. I didn't see other python tests in GenAIExamples, so I wasn't sure if moving the .py test file to `https://github.com/opea-project/GenAIExamples/tree/main/MultimodalQnA/tests` would automatically work or if there would be more work associated with getting GHA to run those python test.

ok, let's move the .py test of `multimodalqna` in GenAIComps  to `GenAIExamples`

File: comps/cores/mega/http_service.py [hunk 4]
```diff
@@ -137,6 +142,34 @@ async def terminate_server(self):
...
+
+    def run(self):
+        """Running method to block the main thread.
+
+        This method runs the event loop until a Future is done. It is designed to be called in the main thread to keep it busy.
+        """
+        self.event_loop.run_until_complete(self.execute_server())
+
+    def start(self, in_single_process=False):
+        if in_single_process:
```
[the remark this answers is not in the document]
Comment (lkk12014402) [reply comps/cores/mega/http_service.py]: remove `multiprocessing.Process`

**Review by lvliang-intel — APPROVED** [review]

**Review by ftian1 — APPROVED** [review]

**Pull request merged**